File: Week-1/Day4/solution.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int nonZero = -1;
        int n = nums.size();
        for(int i=0; i<n; i++) {
            if(!nums[i]) {
                nonZero = nonZero <= i ?  i + 1 : nonZero;
                for(int j=nonZero; j<n; j++) {
                    if( nums[j] ) {
                        swap(nums[i], nums[j]);
                        nonZero = j;
                        break;
                    }
                }
            }
        }
        
    }
};

File: Week-1/Day4/solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        i = 0
        while i<n:
            if nums[i] == 0:
                flag = False
                for j in range(i, n):
                    if nums[j] != 0:
                        temp = nums[i]
                        nums[i] = nums[j]
                        nums[j] = temp
                        flag = True
                        break
                
                if flag == False:
                    break
            else:
                i += 1